lib/Auth/Process/privacyidea.php

```diff
@@ -34,7 +34,13 @@ public function __construct(array $config, $reserved)
         $this->serverconfig['sslverifyhost'] = $cfg->getBoolean('sslverifyhost', null);
         $this->serverconfig['sslverifypeer'] = $cfg->getBoolean('sslverifypeer', null);
         $this->serverconfig['realm'] = $cfg->getString('realm', null);
-        $this->serverconfig['uidKey'] = $cfg->getString('uidKey', null);
+         try {
+             $this->serverconfig['uidKey'] = $cfg->getArray('uidKey');
+             SimpleSAML_Logger::debug("uidArray: try");
+         } catch (Exception $e) {
+             $this->serverconfig['uidKey'] = $cfg->getString('uidKey', null);
```

Comment:
Wouldn't it be more elegant to always write to an array?
So if this is a string, simple write an array with one element.
Then the program code further below does not need to check ``is_array``, it gets a bit simple, since it will always use uidKey[0].

Reply:
Yes, it would be, but it should be backward compatible, shouldn't it?
If we do it always as an array, the old configuration will not work after the update..

Reply:
You misunderstood. Of course the config file itself can contain a string like

~~~~php
    'uidKey' => 'uid',
~~~~
But the config loading code should load this config as array, so that we internally always treat this config as array.

Reply:
please have a look at 84f1c82